internal/rbd/controllerserver.go

```diff
@@ -218,15 +218,23 @@ func checkValidCreateVolumeRequest(rbdVol, parentVol *rbdVolume, rbdSnap *rbdSna
 
 	switch {
 	case rbdSnap != nil:
+
 		err = rbdSnap.isCompatibleEncryption(&rbdVol.rbdImage)
 		if err != nil {
 			return status.Errorf(codes.InvalidArgument, "cannot restore from snapshot %s: %s", rbdSnap, err.Error())
 		}
+		// During CreateSnapshot the rbd image will be created with the
```

Comment:
wow, this is really ugly :cry:

Unfortunatelty I think I understand why this is needed. However, I prefer to have a `isCompatibleThickProvision()` function for the `rbdSnapshot` type instead. That can use the `RbdSnapName` for checking the thick-provisioning state.

(temporary) changing attributes of objects should not be done, it shows lack of proper usage/design of the type in many cases. The `rbdSnapshot` type should probably get corrected, and once that is done, the duplicate `isCompatibleThickProvision()` function can be removed/reconciled.

Reply:
Pushed a new commit with a better approach instead of changing attributes on the snapshot we can generate a new temp volume and use it for checking the metadata and we can destroy it once work is done.

Reply:
@Madhu-1 I am not favoring  the latest approach of  an intermediate volume. The reason being an image creation in the backend just to hide our code issue :) ..

Reply:
@humblec please let me know your suggestion i can do the same.

Reply:
we are not creating an image in the backend it's just a generation of the new temp volume structure.

Reply:
yes, its just a generatino of the new struct, but previously we were exchaning the flag here, we are completely creating a new structure.. I am wondering how this favors the other :)

Reply:
This approach covers the original concern without modifying the original rbdSnapshot struct and reverts it back. the current temp volume generation is already used in other places, not something new.

Reply:
yep, the entire code path need to be revisited in some time. However considering the priority lets survive with this model. Thanks